### PR TITLE
Update vw submodule to commit ae5c4b7d4761f160947f8325c2303589f5b027ff

### PR DESCRIPTION
**Pull request by slahabar**
There are too many commits since the last update on Oct 4th 2021 to note here in the description. 
Please see the files changed.

### CODE REVIEW

**Comment by ataymano** [comment]
Looks like you need to remove & from problematic lines in safe_vw.cc:
Like here:
https://github.com/VowpalWabbit/vowpal_wabbit/commit/1614da485d2d864676e407011c9d69fb2c511f7e

**Comment by olgavrou** [comment]
Changes include:

- logger going from static functions to object
- indicies->indices
- is_multiline -> is_multiline()
- metrics taking a metrics_sink
- some enums are now scoped

**Review by jackgerrits — APPROVED** [review]

File: external_parser/joiners/i_joiner.h [hunk 1]
```diff
@@ -2,14 +2,14 @@
...
+#include "event_processors/reward.h"
 #include "example.h"
-#include "parse_args.h"
 #include "generated/v2/CbEvent_generated.h"
 #include "generated/v2/FileFormat_generated.h"
 #include "generated/v2/Metadata_generated.h"
 #include "lru_dedup_cache.h"
 #include "metrics/metrics.h"
-#include "event_processors/reward.h"
+#include "parse_args.h"
```
Comment (ataymano) [inline external_parser/joiners/i_joiner.h:12]: why do we need this? isn't io/logger.h enough?

Comment (ataymano) [reply external_parser/joiners/i_joiner.h:12]: oh i see it is just moved

**Review by ataymano — APPROVED** [review]

File: external_parser/joiners/i_joiner.h (hunk 1 above)
Comment (olgavrou) [reply external_parser/joiners/i_joiner.h:12]: this wasn't included now, it's just been moved/formatted, it's needed because of `input_options` on line 37

**Pull request merged**